doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -144,61 +144,129 @@ presentation of Cic is given in Chapter :ref:`calculusofinductiveconstructions`.
+                 : `appl_arg`
+   appl_arg : ( `ident` := `term` )
+            : `term9`
+   term9 : .. `term0` ..
+         : `term1`
+   term1 : `term0` .( `qualid` `appl_arg_list_opt` )
+         : `term0` .( @ `qualid` `term9_list_opt` )
+         : `term0` % `ident`
+         : `term0`
+   appl_arg_list_opt : `appl_arg_list_opt` `appl_arg`
+                     : `empty`
+   term9_list_opt : `term9_list_opt` `term9`
+                  : `empty`
+   empty :
+   name_colon : `ident` :
+              : _ :
+   term0 : `atomic_constr`
+         : `match_constr`
+         : ( `term` )
+         : {| `record_fields` | }
+         : `{ `term` }
+         : `( `term` )
+         : ltac : ( `ltac_expr` )
+   record_fields : `record_field_declaration` ; `record_fields`
+                 : `record_field_declaration`
+                 : `record_field` ; `record_fields`
+                 : `record_field` ;
+                 : `record_field`
+                 : `empty`
+   record_field_declaration : `qualid` `binder_list_opt` := `term`
+   record_field : `quoted_attributes_list_opt` `record_binder` `num_opt2` `decl_notation`
+   decl_notation : where `one_decl_notation_list`
+                 : `empty`
+   one_decl_notation_list : `one_decl_notation_list` and `one_decl_notation`
+                          : `one_decl_notation`
+   one_decl_notation : `string` := `constr` `ident_opt3`
+   constr : `term1`
+          : @ `qualid` `instance`
+   ident_opt3 : : `ident`
+              : `empty`
+   record_binder : `name`
+                 : `name` `record_binder_body`
+   record_binder_body : `binder_list_opt` `of_type_with_opt_coercion` `term`
+                      : `binder_list_opt` `of_type_with_opt_coercion` `term` := `term`
+                      : `binder_list_opt` := `term`
+   of_type_with_opt_coercion : :>>
+                             : :> >
+                             : :>
+                             : : > >
+                             : : >
+                             : :
+   num_opt2 : | `num`
+            : `empty`
+   quoted_attributes_list_opt : `quoted_attributes_list_opt` #[ `attribute_list_comma_opt` ]
+                              : `empty`
+   attribute_list_comma_opt : `attribute_list_comma`
+                            : `empty`
+   attribute_list_comma : `attribute_list_comma` , `attribute`
+                        : `attribute`
+   attribute : `ident` `attribute_value`
+   attribute_value : = `string`
+                   : ( `attribute_list_comma_opt` )
+                   : `empty`
+   typeclass_constraint_list_comma : `typeclass_constraint_list_comma` , `typeclass_constraint`
+                                   : `typeclass_constraint`
+   typeclass_constraint : ! `term`
+                        : { `name` } : `exclam_opt` `term`
+                        : `name_colon` `exclam_opt` `term`
+                        : `term`
+   exclam_opt : !
+              : `empty`
+   of_alt : of
+          : &
```

Comment:
This does not seem to be in the right position (no relation to the rest of the grammar chunk).

Reply:
It's referenced by closed_binder above.  Note that I've tried to sort the grammar in depth-first order to the extent possible, with exceptions.  Sometimes definitions are not that close to their uses.

Where do you want me to put it?